src/url.js

```diff
@@ -54,7 +54,11 @@ export function parseUrl(url) {
   };
   // For data URI a.origin is equal to the string 'null' which is not useful.
   // We instead return the actual origin which is the full URL.
-  info.origin = (a.origin && a.origin != 'null') ? a.origin : getOrigin(info);
+  info.origin = (a.origin && a.origin != 'null')
+      ? a.origin
+      : info.protocol == 'data:' || !info.host
+          ? info.href
+          : info.protocol + '//' + info.host;
```

Comment:
I am a big fan of ternary operators. But after a certain point it becomes hard to read, why not use if an else statements?